File: API/Question_Answer/Models/Answer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Question_Answer.Models
{
    public class Answer : Post
    {
        private Question_Answer_DataLayer.Post postDataLayerObject;

        public Answer()
        {
            postDataLayerObject = new Question_Answer_DataLayer.Post();
        }

        public override List<Post> GetPosts(string connnectionString, int maxNumber = 0, string tags = null, int parentId = 0)
        {
            try
            {
                List<Question_Answer_DataLayer.Post> list = postDataLayerObject.GetAnswers(connnectionString, parentId);
                List<Post> answerList = new List<Post>();
                foreach (Question_Answer_DataLayer.Post answer in list)
                {
                    Answer currentPost = (Answer)answer;
                    answerList.Add(currentPost);
                }

                return answerList;
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message);
            }
        }

        

        public override Post AddPost(string connectionString, Post post)
        {
            try
            {
                Question_Answer_DataLayer.Post answerToDataLayer = (Question_Answer_DataLayer.Post)post;
                answerToDataLayer.CreationDate = DateTime.Now;
                answerToDataLayer.LastActivityDate = DateTime.Now;
                Question_Answer_DataLayer.Post resultDataLayer = postDataLayerObject.AddAnswer(connectionString, answerToDataLayer);
                Answer result = (Answer)resultDataLayer;
                return result;
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message);
            }
        }

        public override string DeletePost(string connectionString, int id)
        {
            try
            {
                string result = postDataLayerObject.DeleteAnswer(connectionString, id);
                return result;
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message);
            }
        }

        public override Post UpdatePost(string connectionString, Post post)
        {
            try
            {
                Question_Answer_DataLayer.Post questionToDataLayer = (Question_Answer_DataLayer.Post)post;
                questionToDataLayer.LastActivityDate = DateTime.Now;
                questionToDataLayer.LastEditDate = DateTime.Now;
                Question_Answer_DataLayer.Post resultDataLayer = postDataLayerObject.UpdateAnswer(connectionString, questionToDataLayer);
                Answer result = (Answer)resultDataLayer;
                return result;
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message);
            }
        }

        #region Utilities
        public static explicit operator Answer(Question_Answer_DataLayer.Post q)
        {
            Answer p = new Answer();
            p.Id = q.Id;
            p.AcceptedAnswerId = q.AcceptedAnswerId;
            p.AnswerCount = q.AnswerCount;
            p.Body = q.Body;
            p.CloseDate = q.CloseDate;
            p.CommentCount = q.CommentCount;
            p.CreationDate = q.CreationDate;
            p.FavouriteCount = q.FavouriteCount;
            p.LastActivityDate = q.LastActivityDate;
            p.LastEditDate = q.LastEditDate;
            p.LastEditorDisplayName = q.LastEditorDisplayName;
            p.LastEditorUserId = q.LastEditorUserId;
            p.OwnerUserId = q.OwnerUserId;
            p.ParentId = q.ParentId;
            p.PostTypeId = q.PostTypeId;
            p.Score = q.Score;
            p.Tags = q.Tags;
            p.Title = q.Title;
            p.ViewCount = q.ViewCount;
            return p;
        }
        #endregion
    }
}

File: JAVA API/StackOverflow/src/main/java/com/StackOverflow/StackOverflow/model/Answer.java
package com.StackOverflow.StackOverflow.model;

import java.util.List;

public class Answer extends Post {
    private List<Comment> Comments;

    public Answer(int acceptedAnswerId, int answerCount, String body, String closedDate, int commentCount, String creationDate, int favoriteCount, String lastActivityDate, String lastEditDate, String lastEditorDisplayName, int lastEditorUserId, int ownerUserId, int parentId, int postTypeId, int score, String tags, String title, int viewCount, List<Comment> comments) {
        super(acceptedAnswerId, answerCount, body, closedDate, commentCount, creationDate, favoriteCount, lastActivityDate, lastEditDate, lastEditorDisplayName, lastEditorUserId, ownerUserId, parentId, postTypeId, score, tags, title, viewCount);
        Comments = comments;
    }

    public Answer(int id, int acceptedAnswerId, int answerCount, String body, String closedDate, int commentCount, String creationDate, int favoriteCount, String lastActivityDate, String lastEditDate, String lastEditorDisplayName, int lastEditorUserId, int ownerUserId, int parentId, int postTypeId, int score, String tags, String title, int viewCount) {
        super(id,acceptedAnswerId, answerCount, body, closedDate, commentCount, creationDate, favoriteCount, lastActivityDate, lastEditDate, lastEditorDisplayName, lastEditorUserId, ownerUserId, parentId, postTypeId, score, tags, title, viewCount);
    }

    public Answer(String body, String tags, String title) {
        super(body,tags,title);
    }

    public Answer(String body, String tags, String title, int userId) {
        super(body,tags,title,userId);
    }

    public Answer(int id, String body, String tags, String title, int userId) {
        super(id,body,tags,title, userId);
    }
    public List<Comment> getComments() {
        return Comments;
    }

    public void setComments(List<Comment> comments) {
        Comments = comments;
    }
}
